scaleway/resource_tem_domain.go

```diff
@@ -0,0 +1,219 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	tem "github.com/scaleway/scaleway-sdk-go/api/tem/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayTemDomain() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayTemDomainCreate,
+		ReadContext:   resourceScalewayTemDomainRead,
+		DeleteContext: resourceScalewayTemDomainDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultTemDomainTimeout),
+			Read:    schema.DefaultTimeout(defaultTemDomainTimeout),
+			Delete:  schema.DefaultTimeout(defaultTemDomainTimeout),
+			Default: schema.DefaultTimeout(defaultTemDomainTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The domain name used when sending emails",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the domain",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the domain",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date and time of domain's creation (RFC 3339 format)",
+			},
+			"next_check_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date and time of the next scheduled check (RFC 3339 format)",
+			},
+			"last_valid_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date and time the domain was last found to be valid (RFC 3339 format)",
+			},
+			"revoked_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date and time of the revocation of the domain (RFC 3339 format)",
+			},
+			"last_error": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Error message if the last check failed",
+			},
+			"spf_config": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Snippet of the SPF record that should be registered in the DNS zone",
+			},
+			"dkim_config": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "DKIM public key, as should be recorded in the DNS zone",
+			},
+			"statistics": {
```

Comment:
Not sure about exporting data that are not a state, data will probably change every time we run `terraform apply -refresh`, this is not an expected behavior if resource state has not been changed